lib/salus/scanners/gitleaks.rb

```diff
@@ -0,0 +1,109 @@
+require "salus/scanners/base"
+require "json"
+
+# Gitleaks scanner integration. Detects hardcoded secrets like passwords,
+# api keys, and tokens in git repos.
+# https://github.com/zricethezav/gitleaks
+
+module Salus::Scanners
+  class Gitleaks < Base
+    def run
+      Dir.chdir(@repository.path_to_repo) do
+        copts = config_options
+        shell_return = run_shell(
+          'gitleaks '\
+          '--path . '\
+          '--format sarif '\
+          '--report /dev/stdout '\
+          '--leaks-exit-code 0 ' +
```

Comment:
It seems standard to use 1 as the failure exit code?

Reply:
It appears 0 is instead normally used for success.

Reply:
I'm using `--leaks-exit-code 0` to distinguish between a configuration error with gitleaks and results (which get processed further down).